vm/package/execute.rs

```diff
@@ -0,0 +1,219 @@
+
+use snarkvm_compiler::Execution;
+use snarkvm_console::types::Address;
+
+use super::*;
+
+pub struct ExecutionRequest<N: Network> {
+    execution: Execution<N>,
+    address: Address<N>,
+    program_id: ProgramID<N>,
+}
+
+impl<N: Network> ExecutionRequest<N> {
+    /// Returns a new execution request.
+    pub fn new(execution: Execution<N>, address: Address<N>, program_id: ProgramID<N>) -> Self {
+        Self { execution, address, program_id }
+    }
+
+    /// Sends the request to the given endpoint.
+    pub fn send(&self, endpoint: &str) -> Result<ExecutionResponse<N>> {
+        Ok(ureq::post(endpoint).send_json(self)?.into_json()?)
+    }
+
+    /// Returns the execution.
+    pub const fn execution(&self) -> &Execution<N> {
+        &self.execution
+    }
+
+    /// Returns the program address.
+    pub const fn address(&self) -> &Address<N> {
+        &self.address
+    }
+
+    /// Returns the imports.
+    pub const fn program_id(&self) -> &ProgramID<N> {
+        &self.program_id
+    }
+}
+
+impl<N: Network> Serialize for ExecutionRequest<N> {
+    /// Serializes the deploy request into string or bytes.
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut request = serializer.serialize_struct("ExecutionRequest", 3)?;
+        // Serialize the execution.
+        request.serialize_field("execution", &self.execution)?;
+        // Serialize the address.
+        request.serialize_field("address", &self.address)?;
+        // Serialize the program id.
+        request.serialize_field("program_id", &self.program_id)?;
+        request.end()
+    }
+}
+
+impl<'de, N: Network> Deserialize<'de> for ExecutionRequest<N> {
+    /// Deserializes the deploy request from a string or bytes.
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        // Parse the request from a string into a value.
+        let request = serde_json::Value::deserialize(deserializer)?;
+        // Recover the leaf.
+        Ok(Self::new(
+            // Retrieve the execution.
+            serde_json::from_value(request["execution"].clone()).map_err(de::Error::custom)?,
+            // Retrieve the address of the user executing.
+            serde_json::from_value(request["address"].clone()).map_err(de::Error::custom)?,
+            // Retrieve the program ID.
+            serde_json::from_value(request["program_id"].clone()).map_err(de::Error::custom)?,
+        ))
+    }
+}
+
+pub struct ExecutionResponse<N: Network> {
+    execution: Execution<N>,
+}
+
+impl<N: Network> ExecutionResponse<N> {
+    /// Initializes a new execution response.
+    pub const fn new(execution: Execution<N>) -> Self {
+        Self { execution }
+    }
+
+    /// Returns the execution.
+    pub const fn execution(&self) -> &Execution<N> {
+        &self.execution
+    }
+}
+
+impl<N: Network> Serialize for ExecutionResponse<N> {
```

Comment:
not necessary to solve this in this PR, but I check if we can make this work with a derive or something like that, I'd expect not to have to write custom serializers for simple structs like this one